networking/eth2/src/test/java/tech/pegasys/teku/networking/eth2/rpc/beaconchain/methods/BeaconBlocksByRangeMessageHandlerTest.java

```diff
@@ -269,6 +272,23 @@ public void shouldRejectRequestWhenStepIsZero() {
     verifyNoMoreInteractions(combinedChainDataClient);
   }
 
+  @Test
+  void shouldLimitNumberOfBlocksReturned() {
+    final RequestState requestState =
+        new RequestState(
+            new BeaconBlocksByRangeRequestMessage(
+                UnsignedLong.ZERO, UnsignedLong.valueOf(500), UnsignedLong.ONE),
+            UnsignedLong.valueOf(10000),
```

Comment:
nit: We could make the test more explicit by saying count = Constants.MAX_BLOCK_BY_RANGE_REQUEST_SIZE + 1

Reply:
Done.